st/clitests/awss3api_spec.py

```diff
@@ -1841,6 +1853,335 @@ def get_md5_sum(fname):
+----
+asdf
+bo0
+boo/0...boo/1
+boo#
+boo+
+fo0
+foo/0...foo/2
+foo#123
+foo+123
+qua0
+quax/0...quax/3
+quax#
+quax+
+-----
+'''
+# Step 0: Create directory with above structure.
+# Create temporary 's3listobjvertest' directory
+obj_list = []
+temp_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), "s3listobjvertest")
+os.makedirs(temp_dir, exist_ok=True)
+# Clear contents of 's3listobjvertest' directory
+upload_content = temp_dir + "/*"
+os.system("rm -rf " + upload_content)
+# List of directories to create
+dirs = ['boo', 'foo', 'quax']
+filesize = 1024
+object_range = [2, 3, 4]
+index = 0
+for child_dir in dirs:
+    out_dir = os.path.join(temp_dir, child_dir)
+    os.makedirs(out_dir, exist_ok=True)
+    # Create files within directory
+    for i in range(object_range[index]):
+        filename = str(i)
+        key = "%s/%s" % (child_dir, filename)
+        file_to_create = os.path.join(out_dir, filename)
+        with open(file_to_create, 'wb+') as fout:
+            fout.write(os.urandom(filesize))
+        obj_list.append(key)
+    index = index + 1
+
+# Create remaining root level files
+file_list = ['asdf', 'bo0', 'boo#', 'boo+', 'fo0', 'foo#123', 'foo+123', 'qua0', 'quax#', 'quax+']
+for root_file in file_list:
+    file_to_create = os.path.join(temp_dir, root_file)
+    key = root_file
+    with open(file_to_create, 'wb+') as fout:
+        fout.write(os.urandom(filesize))
+    obj_list.append(key)
+
+# Step 1: Upload folder recrusively.
+#  Step 1.1: Create above keys into bucket 'versionlistbucket'
+AwsTest('Aws Upload folder recursively').upload_objects("versionlistbucket", temp_dir)\
+        .execute_test().command_is_successful()
+
+#  Step 1.2: Create object list file
+object_list_file = create_object_list_file("obj_list.json", obj_list, "true")
+
+# Step 2: Test plain request with no parameters
+# Command:= aws s3api list-object-versions --bucket <bucket>
+# Expected output:
+#   asdf, bo0, boo/0...boo/1, boo#, boo+, fo0, foo/0...foo/2, foo#123
+#   foo+123, qua0, quax/0...quax/3, quax#, quax+
+expected_versions_list = ['asdf', 'bo0', 'boo/0', 'boo/1', 'boo#', 'boo+',
+    'fo0', 'foo/0', 'foo/1', 'foo/2', 'foo#123', 'foo+123', 'qua0', 'quax/0',
+    'quax/1', 'quax/2', 'quax/3', 'quax#', 'quax+']
+list_versions_options = {}
+result = AwsTest('Aws list object versions with no parameters')\
+    .list_object_versions("versionlistbucket", **list_versions_options)\
+    .execute_test()\
+    .command_is_successful()
+
+# Process result set
+list_versions_response = get_aws_cli_object(result.status.stdout)
+if list_versions_response is None:
+    assert False, "Failed to list object versions"
+# Get versions from the response
+object_versions = list_versions_response["versions"]
+print(f"Response: {list_versions_response} Versions received: {object_versions}")
+check = all(item in object_versions for item in expected_versions_list)
+if check is False:
```

Comment:
Are the expected versions not in order as you would expect them? It should be a simple list comparison?

```suggestion
assert object_versions == expected_versions_list, "Failed to match expected versions in the list"
```

Reply:
They should be in the order, doing comparison now.